pkg/kube-metrics/metrics.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubemetrics
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/rest"
+	kcollector "k8s.io/kube-state-metrics/pkg/collector"
+	ksmetric "k8s.io/kube-state-metrics/pkg/metric"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var log = logf.Log.WithName("kubemetrics")
+
+// ServeCRMetrics generates CR specific metrics based on the operator GVK.
+// It starts serving collections of those metrics on given host and port.
+func ServeCRMetrics(cfg *rest.Config, ns []string, operatorTypes map[schema.GroupVersionKind]reflect.Type, host string, port int32) error {
```

Comment:
Since we don't use the `reflect.Type` value of the `operatorTypes` map, I think we should make that parameter type be `[]schema.GroupVersionKind`. That's the intuitive interface for users not using `FilterOutMetaTypes`.

Reply:
That makes more sense now with the below suggestion as well, will do!